client/api_key.go

```diff
@@ -5,11 +5,46 @@ package client
 
 import (
 	"fmt"
+	"github.com/go-openapi/strfmt"
+	log "github.com/sirupsen/logrus"
+	"time"
 
 	clientv1 "github.com/spectrocloud/palette-sdk-go/api/client/v1"
 	"github.com/spectrocloud/palette-sdk-go/api/models"
 )
 
+// CreateAPIKey creates a new API key for the specified tenant.
+func (h *V1Client) CreateAPIKey(name, TenantUID, emailID, org, password string) (string, error) {
+	annotations := map[string]string{
+		"Description": "Automation-Test"}
+	body := &models.V1APIKeyEntity{
+		Metadata: &models.V1ObjectMeta{
+			Name:        name,
+			Annotations: annotations,
+		},
+		Spec: &models.V1APIKeySpecEntity{
+			UserUID: TenantUID,
+			Expiry:  models.V1Time(time.Now().Add(5 * time.Hour)), // Set expiry time
+		},
+	}
+	if _, err := h.AuthLogin(emailID, org, password); err != nil {
```

Comment:
Please remove L30-38. None of this authentication logic belongs here. The user should instantiate their Palette client however they want _before_ calling `CreateAPIKey`

Reply:
yes removed and authenticated before calling CreateAPIKey